src/schemaGenerator.js

```diff
@@ -84,6 +90,164 @@ function addResolveFunctionsToSchema(schema, resolveFunctions) {
   });
 }
 
+class MockList {
+  // wrappedFunction can return another MockList or a value
+  constructor(len, wrappedFunction) {
+    this.len = len;
+    if (typeof wrappedFunction !== 'undefined') {
+      if (typeof wrappedFunction !== 'function') {
+        throw new Error('Second argument to MockList must be a function or undefined');
+      }
+      this.wrappedFunction = wrappedFunction;
+    }
+  }
+
+  mock(o, a, c, r, fieldType, mockTypeFunc) {
+    function randint(low, high) {
+      return Math.floor(Math.random() * (high - low + 1) + low);
+    }
+    let arr;
+    if (Array.isArray(this.len)) {
+      arr = new Array(randint(this.len[0], this.len[1]));
+    } else {
+      arr = new Array(this.len);
+    }
+    for (let i = 0; i < arr.length; i++) {
+      if (typeof this.wrappedFunction === 'function') {
+        const res = this.wrappedFunction(o, a, c, r);
+        if (res instanceof MockList) {
+          arr[i] = res.mock(o, a, c, r, getNullableType(fieldType.ofType), mockTypeFunc);
+        } else {
+          arr[i] = res;
+        }
+      } else {
+        arr[i] = mockTypeFunc(fieldType.ofType)(o, a, c, r);
+      }
+    }
+    return arr;
+  }
+}
+
+function addMockFunctionsToSchema({ schema, mocks, preserveResolvers = false } = {}) {
+  function isObject(thing) {
+    return thing === Object(thing) && !Array.isArray(thing);
+  }
+  // TODO: rewrite from using Map of mock function to using an object?
+  if (!schema) {
+    // XXX should we check that schema is an instance of GraphQLSchema?
+    throw new Error('Must provide schema to mock');
+  }
+  if (!isObject(mocks)) {
+    throw new Error('mocks must be of type Object');
+  }
+
+  // use Map internally, because that API is nicer.
+  const mockFunctionMap = new Map();
+  Object.keys(mocks).forEach((typeName) => {
+    mockFunctionMap.set(typeName, mocks[typeName]);
+  });
+
+  mockFunctionMap.forEach((mockFunction, mockTypeName) => {
+    if (typeof mockFunction !== 'function') {
+      throw new Error(`mockFunctionMap[${mockTypeName}] must be a function`);
+    }
+  });
+
+  const defaultMockMap = new Map();
```

Comment:
I'm curious why you're using a Map rather than a regular JavaScript object for this?